beaconclient/prod_beacon_instance.go

```diff
@@ -257,9 +270,14 @@ func (c *ProdBeaconInstance) GetURI() string {
 }
 
 func (c *ProdBeaconInstance) PublishBlock(block *common.SignedBeaconBlock, broadcastMode BroadcastMode) (code int, err error) {
-	uri := fmt.Sprintf("%s/eth/v2/beacon/blocks?broadcast_validation=%s", c.beaconURI, broadcastMode.String())
+	var uri string
+	if c.ffUseV2PublishBlockEndpoint {
+		uri = fmt.Sprintf("%s/eth/v2/beacon/blocks?broadcast_validation=%s", c.beaconURI, broadcastMode.String())
+	} else {
+		uri = fmt.Sprintf("%s/eth/v1/beacon/blocks", c.beaconURI)
+	}
 	headers := http.Header{}
-	headers.Add("Eth-Consensus-Version", common.ForkVersionStringCapella)
+	headers.Add("Eth-Consensus-Version", common.ForkVersionStringCapella) // optional in v1, required in v2
```

Comment:
at what point should this become Deneb? should we prepare for passing this in as argument?

Reply:
future upgrade path is to add a versioned `block *common.SignedBeaconBlock` struct that will contain the version information. this is implemented in the `Deneb` branch.